### PR TITLE
Default derived constructor uses Rest and Spread argument

### LINKED ISSUE #5147: default derived constructor doesn't spread/rest the args in super call
### Input Code

```js
Array.prototype[Symbol.iterator] = function() {
  throw new Error();
}
class c {} ;
class b extends c {}
new b();
```

### Babel Configuration (.babelrc, package.json, cli command)

presets -latest

### Expected Behavior
```
→ cat test2.js
Array.prototype[Symbol.iterator] = function() {
  throw new Error();
}
class c {} ;
class b extends c {}
var y = new b();

→ ./out.gn/x64.debug/d8 test2.js
test2.js:2: Error
  throw new Error();
  ^
Error
    at Array.(anonymous function) (test2.js:2:9)
    at b (test2.js:5:1)
    at test2.js:6:9
```

### Current Behavior
No error

### Your Environment

| software         | version
| ---------------- | -------
| Babel            |  
| node             | 
| npm              | 
| Operating System |

### PR DESCRIPTION
**Pull request by jridgewell**
Fixes #5147.
Depends on #7675.

### CODE REVIEW

File: packages/babel-plugin-transform-classes/src/transformClass.js [hunk 1]
```diff
@@ -146,20 +146,17 @@ export default function transformClass(
...
     }
-    if (hasConstructor) return;
 
     let params, body;
 
     if (classState.isDerived) {
       const constructor = template.expression.ast`
-        (function () {
-          super(...arguments);
+        (function (...args) {
```
Comment (loganfsmyth) [inline packages/babel-plugin-transform-classes/src/transformClass.js]: An issue description would be helpful. What's the motivation for this change?

Comment (jridgewell) [reply packages/babel-plugin-transform-classes/src/transformClass.js]: Whoops, fixes #5147.

File: packages/babel-plugin-transform-classes/src/transformClass.js [hunk 2]
```diff
@@ -349,20 +346,9 @@ export default function transformClass(
       call = t.logicalExpression("||", bareSuperNode, t.thisExpression());
     } else {
       bareSuperNode = optimiseCall(
```
Comment (Kovensky) [inline packages/babel-plugin-transform-classes/src/transformClass.js]: Looks like the `helper-optimise-call-expression` needs to be updated to support the `super(...args)` case; right now the `optimizeCall` is essentially a noop (or, well, it writes a `.call`, but not an "optimized" one)

File: packages/babel-plugin-transform-classes/test/fixtures/loose/super-class-id-member-expression/output.js [hunk 3]
```diff
@@ -15,8 +15,8 @@ var BaseController2 =
 function (_Chaplin$Controller$A) {
   babelHelpers.inheritsLoose(BaseController2, _Chaplin$Controller$A);
 
-  function BaseController2() {
-    return _Chaplin$Controller$A.apply(this, arguments) || this;
+  function BaseController2(...args) {
+    return _Chaplin$Controller$A.call.apply(_Chaplin$Controller$A, [this].concat(args)) || this;
```
Comment (loganfsmyth) [inline packages/babel-plugin-transform-classes/test/fixtures/loose/super-class-id-member-expression/output.js]: I wonder how this might affect class instantiation performance? My main concern with this change is that it increases output complexity for an extremely edge-casey issue that doesn't seem like anyone would rely on.

Comment (jridgewell) [reply packages/babel-plugin-transform-classes/test/fixtures/loose/super-class-id-member-expression/output.js]: I'm working on a total refactor here, it's just really difficult because the code's a bit low quality at the moment. This will be optimized into a straight `#apply` when I'm done.

Comment (Kovensky) [reply packages/babel-plugin-transform-classes/test/fixtures/loose/super-class-id-member-expression/output.js]: See https://github.com/babel/babel/pull/7678#discussion_r179643407; `helper-optimise-call-expression` just doesn't deal with this corner case and is inserting a `.call` instead of an `.apply`

File: packages/babel-plugin-transform-classes/src/transformClass.js (hunk 1 above)
Comment (nicolo-ribaudo) [reply packages/babel-plugin-transform-classes/src/transformClass.js]: I think that in loose mode we should keep using `arguments`, since it produces a smaller (and probably faster) output.

**Comment by nicolo-ribaudo** [comment]
Closing since tc39/ecma262#2216 got consensus during the last TC39 meeting.
